src/components/entries/NumberField.js

```diff
@@ -103,25 +113,66 @@ export default function NumberFieldEntry(props) {
     setValue,
     step,
     onFocus,
-    onBlur
+    onBlur,
+    validate
   } = props;
 
-  const value = getValue(element);
+  const [ cachedInvalidValue, setCachedInvalidValue ] = useState(null);
+  const globalError = useError(id);
+  const [ localError, setLocalError ] = useState(null);
+
+  let value = getValue(element);
+
+  const previousValue = usePrevious(value);
+
+  useEffect(() => {
+    if (isFunction(validate)) {
+      const newValidationError = validate(value) || null;
+
+      setLocalError(newValidationError);
+    }
+  }, [ value ]);
+
+  const onInput = (newValue) => {
+    let newValidationError = null;
+
+    if (isFunction(validate)) {
+      newValidationError = validate(newValue) || null;
+    }
+
+    if (newValidationError) {
+      setCachedInvalidValue(newValue);
+    } else {
+      setValue(newValue);
```

Comment:
Do we have a test for this? `it('should not set value on error')`.

Reply:
We now do :)